### ISSUE TITLE
Data Ownership TF recommendation - encourage data harmony re netixlan

### ISSUE THREAD

**Issue by ccaputo** [peeringdb/peeringdb#681]
While https://github.com/peeringdb/peeringdb/issues/676 focuses on netixlan dependencies and conflict resolution, the [PeeringDB Data Ownership Policy Document](https://docs.peeringdb.com/gov/misc/2020-04-06_PeeringDB_Data_Ownership_Policy_Document_v1.0.pdf) also encourages data harmony:

```
5.2) Multi-party Data Elements

These data elements can contain data provided by multiple parties. In cases of conflict between
the data sources, the PeeringDB Admin Committee may become involved to help resolve
differences. In addition, the PeeringDB user interface may help guide participants toward
harmony.
```

and

```
6.1) netixlan
[...]
The Task Force recommends PeeringDB employ user interface methods and email notifications to
encourage data harmony between a Network and an Internet Exchange, as a means of expediting
resolution and decreasing the burdens on the Admin Committee.
```

For the case where a Network has 'Allow IXP Update' disabled, this issue has been created to encourage the development of improvements to the user interface that prompt or provide a tip or hint to Networks to add missing data and/or correct data that differs from an IX-F JSON export from an IX.

This matches the goal of data harmony while maintaining the PeeringDB principle that owners decide whether to publish data or not.

**Comment by arnoldnipper** [comment]
Regardless of whether 'Allow IXP Update' is disabled or not there are 'Preview' and 'Post Mortem' buttons. Esp. 'Preview' would show

```
Below is a preview of changes that will be done to your network during the next IX-F data
import. The actual import will run once per day at 00:00Z

We use locally cached IX-F data to generate this preview - if you are expecting changes to an
exchange's IX-F data please give it some time to be reflected here.
```

A network could easily detect if there is an IXP they are connected to but not listening. These messages can also be pulled via an API. However, an email is not sent. I would also not recommend doing so especially when increasing the frequency of the IX-F JSON import.

Was that what you had in mind, @ccaputo? Or is there anything left which should be covered?

**Comment by ccaputo** [comment]
> Was that what you had in mind, @ccaputo? Or is there anything left which should be covered?

I'm thinking of something much more substantial in the user interface.

For example when a user is viewing a 'net' page they have admin rights to, the Public Peering Exchange Points list could include hints showing missing records along with 'Add' and 'Dismiss' buttons beside them, along with a 'Reset Hints' button elsewhere on the page, which would bring back dismissed hints.  Same goes for conflicted records... there could be a 'Resolve Conflict' or 'Make Correction' button.  Both 'Add' and 'Resolve Conflict'/'Make Correction' buttons would result in a similar action to clicking 'Edit' except that they pre-fill out the data or corrections, along with any needed hints/guidance so the user knows what is happening and what will happen when they click Save.

There could even be 'Auto-Add' and 'Auto-Correct' buttons that skip the edit step, assuming the changes that would happen can be made obvious in the UX.  A correction might be as simple as a Speed update or RS Peer status change, or as substantial as an IP assignment change.

IMHO, a user should not have to enable 'Allow IXP Update' in order to benefit from the IX-F JSON data exported by an IX.

@grizz thoughts?

**Comment by arnoldnipper** [comment]
> IMHO, a user should not have to enable 'Allow IXP Update' in order to benefit from the IX-F JSON data exported by an IX.

Do you agree that 'Allow IXP Update' is the most comfortable way to benefit from the IX-F JSON data exported by IXes, @ccaputo?

**Comment by ccaputo** [comment]
> > IMHO, a user should not have to enable 'Allow IXP Update' in order to benefit from the IX-F JSON data exported by an IX.
> 
> Do you agree that 'Allow IXP Update' is the most comfortable way to benefit from the IX-F JSON data exported by IXes, @ccaputo?

Hmm.  It really depends on the network I suppose.  Maybe a better way to phrase what I wrote would be: IMHO, a user should be able to benefit from the IX-F JSON data exported by an IX, even if they choose not to enable 'Allow IXP Update'.

**Comment by arnoldnipper** [comment]
> For example when a user is viewing a 'net' page they have admin rights to, the Public Peering Exchange Points list could include hints showing missing records along with 'Add' and 'Dismiss' buttons beside them,

If we should add this to the list shown in 'Preview' IMO. That seems to be a cleaner approach to me.

**Comment by ccaputo** [comment]
> > For example when a user is viewing a 'net' page they have admin rights to, the Public Peering Exchange Points list could include hints showing missing records along with 'Add' and 'Dismiss' buttons beside them,
> 
> If we should add this to the list shown in 'Preview' IMO. That seems to be a cleaner approach to me.

I feel differently, but mine is just one opinion.  :-)

**Comment by arnoldnipper** [comment]
These are all nice features but have to wait for v3. Maintaining the `netixlan` object becomes more and more complex and perhaps deserves an own page where we are able to present all the options in a clear way. The 'Preview' and 'Post Mortem' lists also would go there as well as drag & drop features to achieve what you called 'auto-add'.  I'm not a UX expert but I feel the current space at "Public Peering Exchange Points" is not enough to place all the buttons and lists.

Thoughts?

**Comment by grizz** [comment]
> > Was that what you had in mind, @ccaputo? Or is there anything left which should be covered?
> 
> I'm thinking of something much more substantial in the user interface.
> 
> For example when a user is viewing a 'net' page they have admin rights to, the Public Peering Exchange Points list could include hints showing missing records along with 'Add' and 'Dismiss' buttons beside them, along with a 'Reset Hints' button elsewhere on the page, which would bring back dismissed hints. Same goes for conflicted records... there could be a 'Resolve Conflict' or 'Make Correction' button. Both 'Add' and 'Resolve Conflict'/'Make Correction' buttons would result in a similar action to clicking 'Edit' except that they pre-fill out the data or corrections, along with any needed hints/guidance so the user knows what is happening and what will happen when they click Save.
> 
> There could even be 'Auto-Add' and 'Auto-Correct' buttons that skip the edit step, assuming the changes that would happen can be made obvious in the UX. A correction might be as simple as a Speed update or RS Peer status change, or as substantial as an IP assignment change.
> 
> IMHO, a user should not have to enable 'Allow IXP Update' in order to benefit from the IX-F JSON data exported by an IX.
> 
> @grizz thoughts?

As I've said before, I agree completely. That sort of feature set could be solved fairly easily with the separate internal tables idea that you proposed earlier.

@arnoldnipper FYI: none of the features you two have discussed would have to wait until v3, they don't remove anything, and, in fact, most are internal data mechanics that any user / API could ignore.

**Comment by arnoldnipper** [comment]
> @arnoldnipper FYI: none of the features you two have discussed would have to wait until v3, they don't remove anything, and, in fact, most are internal data mechanics that any user / API could ignore.

I fully trust you that internally everything is in place to add tons of feature. I always admire how well thought through PDB2.0 is. Awesome work, @grizz and @vegu!

I worry more about the UI/presentation.

**Comment by mcmanuss8** [comment]
Will be implemented in #697
